airflow/decorators/base.py

```diff
@@ -504,6 +504,7 @@ def _expand(self, expand_input: ExpandInput, *, strict: bool) -> XComArg:
             # task's expand() contribute to the op_kwargs operator argument, not
             # the operator arguments themselves, and should expand against it.
             expand_input_attr="op_kwargs_expand_input",
+            map_index_template=task_kwargs.pop("map_index_template", None),
```

Comment:
We might want to get this from `partial_kwargs` instead, which considers `default_args`.